sonar-plugin-api/src/main/java/org/sonar/api/batch/fs/internal/DefaultInputFile.java

```diff
@@ -49,7 +50,7 @@
   private int lastValidOffset = -1;
   private String hash;
   private int nonBlankLines;
-  private int[] originalLineOffsets;
+  private List<Integer> originalLineOffsets;
```

Comment:
I agree with @sns-seb. Keeping the List<Integer> is not a solution. I have reproduced OOM and it happened before the conversion to int[]. The best solution is to use a custom version of ArrayList that will use an int[] internally. See my P/R: https://github.com/SonarSource/sonarqube/pull/1158